Extensions/TileMap/tilemapruntimeobject.js

```diff
@@ -0,0 +1,273 @@
+    newObjectData.content.tilemapAtlasImage
+  ) {
+    this.setTilemapAtlasImage(newObjectData.content.tilemapAtlasImage);
+  }
+  if (oldObjectData.content.displayMode !== newObjectData.content.displayMode) {
+    this.setDisplayMode(newObjectData.content.displayMode);
+  }
+  if (oldObjectData.content.layerIndex !== newObjectData.content.layerIndex) {
+    this.setLayerIndex(newObjectData.content.layerIndex);
+  }
+  if (
+    oldObjectData.content.animationSpeedScale !==
+    newObjectData.content.animationSpeedScale
+  ) {
+    this.setAnimationSpeed(newObjectData.content.animationSpeedScale);
+  }
+  if (
+    oldObjectData.content.animationFps !== newObjectData.content.animationFps
+  ) {
+    this.setAnimationFps(newObjectData.content.animationFps);
+  }
+
+  return true;
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.TileMapRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setHeight(initialInstanceData.height);
+  }
+};
+
+/**
+ * Set the Tilemap json file to display.
+ */
+gdjs.TileMapRuntimeObject.prototype.setTilemapJsonFile = function(tilemapJsonFile) {
+  this._tilemapJsonFile = tilemapJsonFile;
+  this._renderer.updateTileMap();
+};
+
+gdjs.TileMapRuntimeObject.prototype.getTilemapJsonFile = function() {
+  return this._tilemapJsonFile;
+};
+
+gdjs.TileMapRuntimeObject.prototype.isTilemapJsonFileName = function(selectedTilemapJsonFile) {
+  return this._tilemapJsonFile === selectedTilemapJsonFile;
+};
+
+gdjs.RuntimeObject.prototype.setTilesetJsonFile = function(tilesetJsonFile) {
+  this._tilesetJsonFile = tilesetJsonFile;
+  this._renderer.updateTileMap();
+};
+
+gdjs.RuntimeObject.prototype.getTilesetJsonFile = function() {
+  return this._tilesetJsonFile;
+};
+
+gdjs.TileMapRuntimeObject.prototype.setDisplayMode = function(displayMode) {
+  this._displayMode = displayMode;
+  this._renderer.updateTileMap();
+};
+
+gdjs.TileMapRuntimeObject.prototype.getDisplayMode = function() {
+  return this._displayMode;
+};
+
+gdjs.TileMapRuntimeObject.prototype.setLayerIndex = function(layerIndex) {
+  this._layerIndex = layerIndex;
+  this._renderer.updateTileMap();
+};
+
+gdjs.TileMapRuntimeObject.prototype.getLayerIndex = function() {
+  return this._layerIndex;
+};
+gdjs.TileMapRuntimeObject.prototype.setAnimationSpeedScale = function(animationSpeedScale) {
+  this._animationSpeedScale = animationSpeedScale;
+};
+
+gdjs.TileMapRuntimeObject.prototype.getAnimationSpeedScale = function() {
+  return this._animationSpeedScale;
+};
+
+gdjs.RuntimeObject.prototype.setAnimationFps = function(animationFps) {
+  this._animationFps = animationFps;
+};
+
+gdjs.RuntimeObject.prototype.getAnimationFps = function() {
+  return this._animationFps;
+};
+
+gdjs.RuntimeObject.prototype.isTilemapJsonFile = function(
```

Comment:
This function is a duplicate of isTilemapJsonFileName. Remove this one and be sure to review all your functions.

Reply:
good catch. Sorry I forgot to delete it